TU20Bot/Commands/NameMatch.cs

```diff
@@ -0,0 +1,242 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Discord;
+using Discord.Commands;
+using Discord.WebSocket;
+using TU20Bot.Configuration;
+
+namespace TU20Bot.Commands
+{
+    public class NameMatch : ModuleBase<SocketCommandContext>
+    {
+
+        // The max limit of chars set by discord
+        private const int messageLimit = 2000;
+        private Config _config;
+
+        private const int firstNameIndex = 0;
+        private const int lastNameIndex = 1;
+
+        public delegate void IndexMatch(int index);
+        public delegate void IndexMatchDetailed(int index, string firstName, string lastName);
+
+        [Command("lsuser-assignrole")]
+        [Summary("matches guild member names with a specified list and assigns roles to the matched names")]
+        public async Task matchNames(ulong roleId)
+        {
+
+            var users = Context.Guild.Users;
+
+            _config = new Config();
+
+            await sendSplitMessage(await nameMatching(_config.origNames, users, roleId), "\n");
+        }
+
+        public async Task<string> nameMatching(string[,] listNames, IReadOnlyCollection<SocketGuildUser> users, ulong? roleId)
+        {
+
+            StringBuilder errorLog = new StringBuilder("");
+            StringBuilder discordMessage = new StringBuilder("");
```

Comment:
```suggestion
            var errorLog = new StringBuilder();
            var discordMessage = new StringBuilder();
```

Nit pick here really. Feel free to use `var` when you already declare the type of the variable in the expression. Saves you some typing time and looks a bit nicer imo.